### DISCUSSION TITLE
Rustdesk Pro Server in HA mode

### DISCUSSION THREAD

**Discussion by BlockheadVFX-IT** [rustdesk/rustdesk-server-pro#255] [General]
Hi All

Was just wandering if RustDesk server pro can be self hosted in HA mode? I understand there are relay server option to use to connect to devices for less latency. I was just thinking we have  signalling server in HA mode , so if one server goes down in one location, another server can catch-up.

We have offices in multiple location ( Auckland and Sydney), and our signalling server is in Auckland. We have relay server in Sydney so people in Sydney use relay server for better connection.

Last week we had some issues with the signalling server VM so it was shut. and of course we were not able to make connections. 

So I was wandering if I can setup signalling server in each location (either syncing with each other or something), so we can still operate in those scenarios. 

your thoughts and suggestions are welcomed. 

Thank you

**Comment by rustdesk** [comment]
I would suggest you set up your backup/restore strategy, do not trust any other 3rd HA solution, we have been seeing so many failures of biggest companies in the world every month.

You can write some cron job to backup files repeatly, https://github.com/rustdesk/rustdesk-server-pro/discussions/184#discussioncomment-8013375. Once your hbbs server fails, you can start a new hbbs with the backup file, migrate the license to the new server, modify dns to new ip, after dns ttl time, your rustdesk clients will connect to new hbbs automatically.

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk-server-pro/discussions/184
